### ISSUE TITLE
Deserializer failing with specific object

### ISSUE THREAD

**Issue by propagated** [restsharp/RestSharp#607]
I am using `Execute<T>(request)` to deserialize. My object model looks like this:

```
    public class FilesItem
    {
        public String name { get; set; }
        public Boolean isEmail { get; set; }
        public Dictionary<string,string> oEmailData { get; set; }
        public String test { get; set; }
    }
    public class FilesList
    {
        public List<FilesItem> files { get; set; }
    }
```

My server is generating the following json to match:
`{"files":[{"name":"ea5e83d2-ec57-4720-94a0-6228aa0cbf33.pdf","isEmail":false,"oEmailData":null,"test":null}]}`

`client.Execute<FilesList>(request).Data` is coming back null every time. Feels like a bug somewhere in dealing with Dictionary<T,T> or the fact that i'm getting back nulled data? Both scenarios are supported in the RS docs.

**Comment by ps2goat** [comment]
I'm facing the same issue.  You could at least provide a way for us to set the deserializer so we could write our own implementation that uses JSON.NET.  You have the IDeserializer interface with no apparent way to force the Execute(Of T) to use a custom one.

**Comment by hallem** [comment]
@ps2goat there is a way to use the JSON.NET library with rest sharp.  If you look at the readme there are some quick instructions on how to do it there, let me know if you need more help beyond that.

**Comment by ps2goat** [comment]
@hallem, I already implemented the JSON.NET library, but I wanted to be able to disconnect it from my project and be able to provide an interface implementation to RestSharp.  It already has the Execute(Of T), so I wanted to use that.  My implementation is still clean, but it doesn't adhere to the Onion model as well as it should.

**Comment by alexeyzimarev** [comment]
Serialisation using JSON.NET will be made easily available by #1018
